sympy/solvers/solveset.py

```diff
@@ -1754,6 +1768,46 @@ def _is_logarithmic(f, symbol):
     return rv
 
 
+def _compute_lambert_solutions(lhs, rhs, symbol):
+    r"""
+    Computes the equations whose solutions will be given in terms
+    of the list of LambertW function. Returns `None` if it fails doing so.
+
+    Example
+    =======
+
+    >>> from sympy.solvers.solveset import _compute_lambert_solutions
+    >>> from sympy import symbols, exp
+    >>> x = symbols('x')
+    >>> _compute_lambert_solutions(x*exp(x), 1, x)
+    [LambertW(1)]
+
+    """
+    try:
+        poly = lhs.as_poly()
+        gens = _filtered_gens(poly, symbol)
+        return _solve_lambert(lhs - rhs, symbol, gens)
+    except NotImplementedError:
+        pass
```

Comment:
Which of the three lines above can raise `NotImplementedError`?

Only one of these should be inside the `try` I think.

Reply:
```suggestion
    poly = lhs.as_poly()
    gens = _filtered_gens(poly, symbol)
    try:
        return _solve_lambert(lhs - rhs, symbol, gens)
    except NotImplementedError:
        pass
```
What about this ?

Reply:
Yes, that's better